src/nbl/ui/CWindowWin32.cpp

```diff
@@ -485,6 +513,132 @@ namespace ui
 		return DefWindowProc(hWnd, message, wParam, lParam);
 	}
 
+	E_KEY_CODE CWindowWin32::getNablaKeyCodeFromNative(uint32_t nativeWindowsKeyCode)
+	{
+		nbl::ui::E_KEY_CODE nablaKeyCode = EKC_NONE;
+		switch (nativeWindowsKeyCode)
+		{
+		case VK_BACK:			nablaKeyCode = EKC_BACKSPACE; break;
+		case VK_TAB:			nablaKeyCode = EKC_TAB; break;
+		case VK_CLEAR:			nablaKeyCode = EKC_CLEAR; break;
+		case VK_RETURN:			nablaKeyCode = EKC_ENTER; break;
+		case VK_SHIFT:			[[fallthrough]];
+		case VK_LSHIFT:			[[fallthrough]];
+		case VK_RSHIFT:			nablaKeyCode = EKC_SHIFT; break;
+		case VK_CONTROL:		[[fallthrough]];
+		case VK_LCONTROL:		[[fallthrough]];
+		case VK_RCONTROL:		nablaKeyCode = EKC_CONTROL; break;
+		case VK_LMENU:			[[fallthrough]];
+		case VK_RMENU:			[[fallthrough]];
+		case VK_MENU:			nablaKeyCode = EKC_ALT; break;
+		case VK_PAUSE:			nablaKeyCode = EKC_PAUSE; break;
+		case VK_CAPITAL:		nablaKeyCode = EKC_CAPS_LOCK; break;
+		case VK_ESCAPE:			nablaKeyCode = EKC_ESCAPE; break;
+		case VK_SPACE:			nablaKeyCode = EKC_SPACE; break;
+		case VK_PRIOR:			nablaKeyCode = EKC_PAGE_UP; break;
+		case VK_NEXT:			nablaKeyCode = EKC_PAGE_DOWN; break;
+		case VK_END:			nablaKeyCode = EKC_END; break;
+		case VK_HOME:			nablaKeyCode = EKC_HOME; break;
+		case VK_LEFT:			nablaKeyCode = EKC_LEFT_ARROW; break;
+		case VK_RIGHT:			nablaKeyCode = EKC_RIGHT_ARROW; break;
+		case VK_UP:				nablaKeyCode = EKC_UP_ARROW; break;
+		case VK_DOWN:			nablaKeyCode = EKC_DOWN_ARROW; break;
+		case VK_SELECT:			nablaKeyCode = EKC_SELECT; break;
+		case VK_PRINT:			nablaKeyCode = EKC_PRINT; break;
+		case VK_EXECUTE:		nablaKeyCode = EKC_EXECUTE; break;
+		case VK_SNAPSHOT:		nablaKeyCode = EKC_PRINT_SCREEN; break;
+		case VK_INSERT:			nablaKeyCode = EKC_INSERT; break;
+		case VK_DELETE:			nablaKeyCode = EKC_DELETE; break;
+		case VK_HELP:			nablaKeyCode = EKC_HELP; break;
+		case 0x30:				nablaKeyCode = EKC_0; break;
```

Comment:
why raw numbers instead of `VK_` constants?

Reply:
bc winapi has no `VK_` constants for that, just numbers: https://docs.microsoft.com/en-us/windows/win32/inputdev/virtual-key-codes

Reply:
but the case label should be `0` cause 0x30 is ascii code for 0